src/cursor.js

```diff
@@ -10,8 +10,21 @@
 import {valToKeyPath, makeCursor} from './utils';
 import createAtom from 'atom-store';
 
+const inMemory = (initial, transition) => {
+  let rootState = initial;
+  const read = () => rootState;
+  const write = (fn, ...context) => {
+    const oldState = read();
+    const newState = fn(oldState);
+    transition(newState, oldState, ...context);
```

Comment:
Great. I'd be keen to wait for this to go into `atom-store`, so that `immutable-cursor` can rely on the core revised `inMemory` plugin.